sdk/core/azure-json/src/main/java/com/azure/json/JsonObject.java

```diff
@@ -0,0 +1,449 @@
+package com.azure.json;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Iterator;
+
+/**
+ * Class representing the JSON object type.
+ */
+public class JsonObject extends JsonDataStructure {
+
+    public JsonObject() {
+    }
+
+    /**
+     * Constructor specifically used to invoke a build from a reader. It
+     * @param reader is any currently open JsonReader object. This is utilised by JsonBuilder and also
+     *               called recursively by objects. If the developer knows they want to build an object
+     *               from a JSON, then they can bypass the JsonBuilder and just use this constructor.
+     */
+    public JsonObject(JsonReader reader) {
+        try {
+            this.build(reader);
+        } catch (Exception e) {
+            // todo sort this logging out so it's not just printing a stack trace...
+            e.printStackTrace();
+        }
+    }
+
+
+    /**
+     * Stores the key and values of each property in the JsonObject. The values
+     * may be any valid JSON type: object, array, string, number, boolean, and null
+     */
+
+    private Map<String, JsonElement> properties = new LinkedHashMap<>();
```

Comment:
Nice! I'm glad to see `LinkedHashMap` here as it's the right choice to maintain element ordering